File: src/main/java/org/goafabric/personservice/repository/entity/PersonEo.java
package org.goafabric.personservice.repository.entity;

import jakarta.persistence.*;
import org.goafabric.personservice.repository.extensions.AuditListener;
import org.hibernate.annotations.GenericGenerator;

@Entity
@Table(name = "person")
@EntityListeners(AuditListener.class)
public class PersonEo {
    @Id
    @GeneratedValue(generator = "uuid") @GenericGenerator(name = "uuid", strategy = "uuid2")
    public String id;

    public String firstName;

    public String lastName;

    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    public AddressEo address;

}

File: doc/examples/discriminator/PersonEo.kt
package org.goafabric.personservice.persistence.entity

import jakarta.persistence.*
import org.goafabric.personservice.persistence.extensions.AuditTrailListener
import org.goafabric.personservice.persistence.extensions.KafkaPublisher
import org.hibernate.annotations.TenantId

@Entity
@Table(name = "person")
@EntityListeners(AuditTrailListener::class, KafkaPublisher::class)
class PersonEo (
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: String?,

    var organizationId: String?,

    @TenantId
    var tenantId: String?,

    var firstName: String,
    var lastName: String,

    @OneToMany(cascade = [CascadeType.ALL])
    @JoinColumn(name = "person_id")
    var address: MutableList<AddressEo> = mutableListOf(),

    @Version //optimistic locking
    var version: Long
)
